### PR TITLE
[Docs] Cleanup Cog Guide

### PR DESCRIPTION
**Pull request by bobloy**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes

Adds the cleanup cog guide to the docs. Based on the descriptions provided in the commands.
Personalization may be necessary, feedback on that would be helpful.

### CODE REVIEW

**Comment by bobloy** [comment]
The strategy I took was generating the docs based on the cog, so this PR also includes updates to the cleanup cog.
If this strategy is approved, I'll add PRs for the remaining cogs.

**Comment by bobloy** [comment]
Found the related issue: #1734

File: docs/cog_guides/cleanup.rst [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+**Syntax**
+
+.. code-block:: none
+
+    [p]cleanup before <message_id> <number> [delete_pinned=False]
+
+**Description**
+
+Deletes X messages before specified message.
```
Comment (Flame442) [inline docs/cog_guides/cleanup.rst]:
```suggestion
Deletes X messages before the specified message.
```
(Although I know this isn't an issue directly from you, it should get changed here and in [p]cleanup before's docstring)

**Review by Flame442 — CHANGES_REQUESTED** [review]
Just a minor grammar fix I would like to get handled before it spreads

File: docs/cog_guides/cleanup.rst (hunk 1 above)
Comment (bobloy) [reply docs/cog_guides/cleanup.rst]: Fixed with commit

**Review by Flame442 — APPROVED** [review]
Thanks!

**Pull request merged**